arch/arm64/core/mmu.c

```diff
@@ -629,7 +630,15 @@ static int __add_map(struct arm_mmu_ptables *ptables, const char *name,
 	__ASSERT(((virt | phys | size) & (CONFIG_MMU_PAGE_SIZE - 1)) == 0,
 		 "address/size are not page aligned\n");
 	desc |= phys;
-	return set_mapping(ptables, virt, size, desc, may_overwrite);
+	ret = set_mapping(ptables, virt, size, desc, may_overwrite);
+	if (ret) {
+		return ret;
+	}
+
+	if (MT_TYPE(attrs) == MT_NORMAL || MT_TYPE(attrs) == MT_NORMAL_WT) {
```

Comment:
only add this condition + change description of commit message related to the below line